spyder/plugins/workingdirectory/plugin.py

```diff
@@ -4,232 +4,165 @@
-        """Load history from a text file in user home directory"""
-        if osp.isfile(self.LOG_PATH):
-            wdhistory, _ = encoding.readlines(self.LOG_PATH)
-            wdhistory = [name for name in wdhistory if os.path.isdir(name)]
+
+    def get_description(self):
+        return _('Set current working directory on different spyder plugins')
+
+    def get_icon(self):
+        return self.create_icon('DirOpenIcon')
+
+    def register(self):
+        container = self.get_container()
+        editor = self.get_plugin(Plugins.Editor)
+        explorer = self.get_plugin(Plugins.Explorer)
+        ipyconsole = self.get_plugin(Plugins.IPythonConsole)
+
+        self.add_application_toolbar(self.NAME, container.toolbar)
+        container.sig_current_directory_changed.connect(
+            self.sig_current_directory_changed)
+        self.sig_current_directory_changed.connect(
+            lambda path, plugin=None: self.chdir(path, plugin))
+        container.set_history(self.load_history())
+
+        if editor:
+            editor.sig_dir_opened.connect(
+                lambda path, plugin=editor: self.chdir(path, editor))
+
+        if ipyconsole:
+            self.sig_current_directory_changed.connect(
+                ipyconsole.set_current_client_working_directory)
+            # TODO: chdir_current_client might follow a better naming
+            # convention
+            ipyconsole.sig_current_directory_changed.connect(
+                lambda path, plugin=ipyconsole: self.chdir(path, plugin))
+
+        if explorer:
+            self.sig_current_directory_changed.connect(
+                lambda path: explorer.chdir(path, emit=False))
+            explorer.sig_dir_opened.connect(
+                lambda path, plugin=explorer: self.chdir(path, plugin))
+
+    # --- Public API
+    # ------------------------------------------------------------------------
+    def chdir(self, directory, sender_plugin=None):
+        """
+        Change current working directory.
+
+        Parameters
+        ----------
+        directory: str
+            The new working directory to set.
+        sender_plugin: spyder.api.plugins.SpyderPluginsV2
+            The plugin that requested this change: Default is None.
+        """
+        explorer = self.get_plugin(Plugins.Explorer)
+        ipyconsole = self.get_plugin(Plugins.IPythonConsole)
+
+        if explorer and sender_plugin != explorer:
+            explorer.chdir(directory, emit=False)
+            explorer.refresh_plugin(directory, force_current=True)
+
+        if ipyconsole and sender_plugin != ipyconsole:
+            ipyconsole.set_current_client_working_directory(directory)
+
+        if sender_plugin is not None:
+            container = self.get_container()
+            container.chdir(directory, emit=False)
+
+        self.save_history()
+
+    def load_history(self, workdir=None):
+        """
+        Load history from a text file in user home directory and sets the

```

Comment:
This docstring is wrong because it's loading history from a file in our config system.

Reply:
Fixed